### ISSUE TITLE
BUG: Soft-whitelisted mods are ignored after the first connection

### ISSUE THREAD

**Issue by losscut** [DiffuseHyperion/InertiaAntiCheat#19]
**Describe the bug**
Client disconnects after reconnect due to an invalid hashsum, although all mods match the configuration.
It seems that anticheat stops taking into account mods installed on the client, starting from the first connection. Two specific mods with hashes 1b31217564427ce561c6ca77fdfa3466 (Litematica) and 5db8fbcf75e3eb2b81e67aed577ab579 (Malilib), which were installed on the client, are in the whitelist at the first connection, but missing at the second (hence are included in combined hash).
[first-connection.txt](https://github.com/DiffuseHyperion/InertiaAntiCheat/files/14178657/first-connection.txt)
[second-connection.txt](https://github.com/DiffuseHyperion/InertiaAntiCheat/files/14178668/second-connection.txt)

**To Reproduce**
Steps to reproduce the problem:
1. Launch the game with optional mods
2. Connect to server
3. Disconnect
4. Connect to server once again
5. Disconnection due to failed modlist check

**Expected behavior**
Allowed connection with soft-whitelisted mods

**Server information:**
 - Server software: Fabric
 - Server software loader version: 0.15.6
 - Minecraft version: 1.20.4
 - Mod version: 1.0.0
 - Fabric API version: 0.95.4+1.20.4

**Mod config**
```
[mods]
method = "group"
algorithm = "MD5"
vanillaKickMessage = "No anticheat"
deniedKickMessage = "Wrong modpack"

[mods.individual]
blacklist = []
friendlyBlacklist = []
whitelist = []
friendlyWhitelist = []

[mods.group]
checksum = ["a2e13abe3da24471639ffd773ff57f66"]
friendlyChecksum = ["Current Modpack: ", "Livermorium 1.5.0"]
softWhitelist = ["8281fecec375367802e5bba2d5b08d68", "61c622d98e5938c377a45ad513b6de13", "d8d32fc651bdad634b10ac94d0f35c02", "e3fc116a15f56a6dd7840e20ee9a14c8", "ae0360b5a3dacbb4a32008ea7838040f", "017e8bc6881bd423575dd6576721d798", "1b31217564427ce561c6ca77fdfa3466", "5db8fbcf75e3eb2b81e67aed577ab579", "f0709ed3d764e564fde6d4b788fd5257"]

[debug]
debug = true
version = 5
```

**Full logs**
https://mclo.gs/sWlzkxn - server log with 5 connections
https://mclo.gs/rjxBXav - client log with the 1st and 2nd connections
https://mclo.gs/fUYwcpS - 3rd connection (error on line 945 refers to raknetify, which is disabled on the server)
https://mclo.gs/vB8oDSw - 4th and 5th connections

**Comment by DiffuseHyperion** [comment]
Sorry for taking so long to notice, the issue should be fixed. Will be pushing a new release momentarily.

**Comment by DiffuseHyperion** [comment]
Just released [1.0.1](https://github.com/DiffuseHyperion/InertiaAntiCheat/releases/tag/1.0.1), which fixes the bug. Thanks for reporting :)

**Comment by Articulless** [comment]
бля соси лох
